### PR TITLE
Use a predefined color when authorInfo.colorId is not defined (fixes #3719)

### LINKED ISSUE #3719: Cannot read property 'colorId' of null
```
Mar 16 09:27:17 run.sh[1746]: [2020-03-16 09:27:17.291] [ERROR] console - (node:1746) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'colorId' of null
Mar 16 09:27:17 run.sh[1746]: at /home/etherpad-lite/src/node/handler/PadMessageHandler.js:1199:37
Mar 16 09:27:17 run.sh[1746]: at runMicrotasks (<anonymous>)
Mar 16 09:27:17 run.sh[1746]: at processTicksAndRejections (internal/process/task_queues.js:97:5)
Mar 16 09:27:17 run.sh[1746]: at async Promise.all (index 0)
Mar 16 09:27:17 run.sh[1746]: at async handleClientReady (/home/etherpad-lite/src/node/handler/PadMessageHandler.js:1171:5)
Mar 16 09:27:17 run.sh[1746]: [2020-03-16 09:27:17.291] [ERROR] console - (node:1746) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 76)
Mar 16 09:27:19 run.sh[1746]: [2020-03-16 09:27:19.034] [WARN] message - Dropped message, USERINFO_UPDATE Session not ready.[object Object]
```
https://github.com/ether/etherpad-lite/blob/develop/src/node/handler/PadMessageHandler.js#L1199

### PR DESCRIPTION
**Pull request by JohnMcLear**
cc @muxator

### CODE REVIEW

**Comment by JohnMcLear** [comment]
@muxator Ready to go.

**Comment by muxator** [comment]
Note that the error in #3719 is die to a missing `authorInfo`.

This PR in that case would not set `colorId` to the fixed value `#daf0b2`, but would simply terminate the callback chain. Maybe this is the behaviour you wanted, but it's worth pointing it. Do you agree?

This is introduced in your original commit "slightly better error handling", originally here: 3cdd8b71b1997b4252f6914c51e89c080c56822b.

That commit is currently rewritten to this one (it's logically equivalent, just uses an early return. Maybe it's easier to spot): 27a179daba966b88b805e2fbb13ffdc3f6ce1aad

**Comment by JohnMcLear** [comment]
Yea, so I went a bit cleaner, I know the code is mheh but remember this is  more of a catch case which will allow for further debug as hopefully someone can provide steps to replicate consistently.

File: src/node/handler/PadMessageHandler.js [hunk 1]
```diff
@@ -1189,10 +1189,10 @@ async function handleClientReady(client, message)
       let p = cached ? Promise.resolve(cached) : authorManager.getAuthor(author);
 
       return p.then(authorInfo => {
+
         if (!authorInfo) {
           console.warn("For some reason Etherpad could not find the authorInfo for this editor, see https://github.com/ether/etherpad-lite/issues/3719");
-
-          return;
+          let authorInfo = {};
```
[the remark this answers is not in the document]
Comment (muxator) [reply src/node/handler/PadMessageHandler.js]: I have removed the let statement.

**Comment by muxator** [comment]
I will also reverse the order of the two checks, so when there isn't if there is no color, the log message at least contains an author name (or "anonymous").

File: src/node/handler/PadMessageHandler.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (JohnMcLear) [reply src/node/handler/PadMessageHandler.js]: ah good shout.

**Pull request merged**